sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs

```diff
@@ -151,11 +151,17 @@ private IEnumerable<string> FindCredentialWords(string variableName, string vari
                     return Enumerable.Empty<string>();
                 }
 
-                var credentialWordsFound = variableName
+                var credentialWordsFound = (variableName ==null || variableName.Any(x => char.IsLower(x)) ? variableName : variableName.ToLower()) // Prepare "PASSWORD" for SplitCamelCaseToWords()
```

Comment:
I think this fix deserves a separate ticket: fixing camelcase split for fully uppercase words should be directly done inside the `SplitCamelCaseToWords` method IMO.
But since it would impact other part of the project, I suggest to do it in a separate commit.
WDYT ?